File: kotlin/src/main/kotlin/advertisements/shared/value_object/CivicCenterId.kt
package advertisements.shared.value_object

import advertisements.shared.exceptions.InvalidUniqueIdentifierException

class CivicCenterId private constructor(private val value: String) {

    init {
        if (!validate(value)) {
            throw InvalidUniqueIdentifierException.withId(value)
        }
    }

    fun value(): String {
        return value
    }

    fun equals(id: CivicCenterId): Boolean {
        return this.value == id.value
    }

    private fun validate(value: String): Boolean {
        val regex = Regex("^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$", RegexOption.IGNORE_CASE)
        return regex.matches(value)
    }

    companion object {
        @Throws(InvalidUniqueIdentifierException::class)
        fun create(value: String): CivicCenterId {
            return CivicCenterId(value)
        }
    }
}

File: php/src/Advertisements/Shared/Domain/ValueObjects/CivicCenterId.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\Shared\Domain\ValueObjects;

use Demo\App\Advertisements\Shared\Domain\Exceptions\InvalidUniqueIdentifierException;

final readonly class CivicCenterId
{
    /**
     * @throws InvalidUniqueIdentifierException
     */
    public function __construct(private string $value)
    {
        if (!$this->validate($value)) {
            throw InvalidUniqueIdentifierException::withId($this->value);
        }
    }

    public function value(): string
    {
        return $this->value;
    }

    public function equals(CivicCenterId $id): bool
    {
        return $this->value === $id->value;
    }

    private function validate(string $value): bool
    {
        return preg_match('/^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i', $value) != 0;
    }
}
